core/src/test/scala/org/apache/spark/util/JsonProtocolSuite.scala

```diff
@@ -432,6 +439,25 @@ class JsonProtocolSuite extends SparkFunSuite {
     testAccumValue(Some("anything"), 123, JString("123"))
   }
 
+  test("ExecutorMetrics backward compatibility") {
+    // ExecutorMetrics is newly added
+    val accumUpdates =
+      makeTaskMetrics(300L, 400L, 500L, 600L, 700, 800, hasHadoopInput = true, hasOutput = true)
+        .accumulators().map(AccumulatorSuite.makeInfo)
+        .zipWithIndex.map { case (a, i) => a.copy(id = i) }
+    val executorMetricsUpdate = SparkListenerExecutorMetricsUpdate("exec3", new ExecutorMetrics,
+      Seq((1L, 2, 3, accumUpdates)))
+    assert(executorMetricsUpdate.executorMetrics != null)
+    assert(executorMetricsUpdate.executorMetrics.transportMetrics != null)
+    val newJson = JsonProtocol.executorMetricsUpdateToJson(executorMetricsUpdate)
+    val oldJson = newJson.removeField { case (field, _) => field == "Executor Metrics Updated"}
+    val newMetrics = JsonProtocol.executorMetricsUpdateFromJson(oldJson)
+    assert(newMetrics.executorMetrics.hostname === "")
```

Comment:
it seems like really `executorMetrics` should be an `Option` (with a comment that it was not present till version x)